client_integration_test.go

```diff
@@ -1489,6 +1489,49 @@ func TestClientReadFrom(t *testing.T) {
 	}
 }
 
+func TestClientReadFromSectionReader(t *testing.T) {
+	const (
+		packetSize = 1024
+		bufsize    = 4 * packetSize
+		skip       = 100
+		filesize   = bufsize - skip
+	)
+
+	sftp, cmd := testClient(t, READWRITE, NODELAY,
+		MaxPacketChecked(packetSize),
+		UseConcurrentWrites(true))
+	defer cmd.Wait()
+	defer sftp.Close()
+
+	d, err := ioutil.TempDir("", "sftptest-readfrom-sectionreader")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(d)
+
+	buf := bytes.NewReader(make([]byte, bufsize))
+	r := io.NewSectionReader(buf, 0, bufsize)
+
+	// Skip some bytes so that r.Size is no longer accurate.
+	// ReadFrom uses it to determine concurrency,
+	// but should not rely on it to determine how many bytes to send.
+	r.Seek(skip, io.SeekStart)
```

Comment:
We should be able to do this with the `bytes.Reader` itself. https://pkg.go.dev/bytes@go1.17.5#Reader.Size

Reply:
You mean not use a SectionReader at all?

Reply:
yeah, we should be able to do the same test with `bytes.Reader` and not need `SectionReader`

Reply:
That's less strong a test, though, since the type switch inside ReadFrom first checks if its Reader has a Len method, and bytes.Reader does:

https://github.com/pkg/sftp/blob/1849af66afae504778b257e286cd1a406ea14bc5/client.go#L1770-L1785

Reply:
Ach so. So the SectionReader is to get a `Size, but not a `Len`?` 🤔 I think things would be clearer if we use a one off type to make it clear what exact functionality we’re testing against. That is, don’t require people to look up `SectionReader` to know exactly what functionality we’re testing against.

Reply:
Good idea. Done.